apiserver/apiserver/apiserver.py

```diff
@@ -19,6 +20,26 @@
 routes = web.RouteTableDef()
 
 
+with open(os.environ.get('HAIL_JWT_SECRET_KEY_FILE') or '/jwt-secret/secret-key') as f:
+    jwtclient = hj.JWTClient(f.read())
+
+
+def authenticated_users_only(fun):
+    def wrapped(request, *args, **kwargs):
+        encoded_token = request.cookies.get('user')
+        if encoded_token is not None:
+            try:
+                userdata = jwtclient.decode(encoded_token)
+                if 'userdata' in fun.__code__.co_varnames:
+                    return fun(request, *args, userdata=userdata, **kwargs)
+                return fun(request, *args, **kwargs)
+            except jwt.exceptions.DecodeError:
+                pass
+        raise web.HTTPUnauthorized(headers={'WWW-Authenticate': 'Bearer'})
```

Comment:
I think we’re not using Authorization Bearer tokens, but rather a cookie

I think something along the lines of : ‘Cookie’

There’s also a realm property I’m just reading about

Reply:
There's an [official list of schemes](https://www.iana.org/assignments/http-authschemes/http-authschemes.xhtml). There does not appear to be one for JWT tokens. JWT tokens meet the definition of "bearer tokens" given by the OAuth 2.0 RFC:

> Any party in possession of a bearer token (a "bearer") can use it to get access to the associated resources (without demonstrating possession of a cryptographic key).  To prevent misuse, bearer tokens need to be protected from disclosure in storage and in transport.

Not sure how much it matters. GitHub doesn't even send this header.

Reply:
You're right the Bearer is the most appropriate, thanks!